### ISSUE TITLE
Add “Curly quotes” option or plugin

### ISSUE THREAD

**Issue by rgreen5** [musescore/MuseScore#17550]
### Your idea

Request for either (1) A _native_ plugin to automatically convert straight quotes in the score (single and double) to curly (smart) quotes. This would work on all text objects including lyrics attached to notes; or (2) a global option (like LibreOffice) so that quotes (single and double) typed in from the keyboard are automatically entered as the curly variety.

**Note**: The current "Curly Quotes" plugin does not work, but crashes the program

### Problem to be solved

The ability to convert the standard keyboard straight quotes to the curly (smart) variety is essential for professional printed output. Currently you can only do this by using the special characters dialog, or pasting in from an external character map.

**Comment by Jojo-Schmitz** [comment]
The current Curly Quotes plugin is for MuseScore 3, it crashes in MuseScore 4 because it (still) uses `Qt.quit()`: https://github.com/dbolton/curlyQuotes/blob/970b16b6323202eb314a196c59d7b8cb4078d74f/curlyQuotes.qml#L10 and https://github.com/dbolton/curlyQuotes/blob/970b16b6323202eb314a196c59d7b8cb4078d74f/curlyQuotes.qml#L87
Should be easy to fix.
If I'd only get that plugin to work in Mu3 in the first place..., @dbolton?

**Comment by Jojo-Schmitz** [comment]
BTW: Curly quotes depend on locale/language, e.g. in German(y) we use „bla bla“ rather than „bla bla” and ‚bla‘ rather than ‘bla’

**Comment by dbolton** [comment]
@Jojo-Schmitz Thanks, I approved your edits and released the update. 

English computer keyboards only have straight quotes, which means software has to automatically turn the quotes curly. If you are interested, we could potentially release a German version of Curly Quotes. Would the expectation be that it convert "bla bla" to „bla bla“? Is there an equivalent list of German contractions that start with an apostrophe? (See line 22 of curlyQuotes.qml for my list of English words.) 

Note: in MuseScore 4 it looks like you have choose Plugins > Curly Quotes then double click on any lyric to see the change take affect throughout the score.

**Comment by Jojo-Schmitz** [comment]
German keybords suffer from the same issue. So it is up to the software to auto-correct this. MS Office (Word, Outlook, etc.) does that, changing “bla bla”  to „bla bla“ and ‘bla’ to ‚bla‘
There maybe colloquial German where those contractions are used, from the top of my head I can't come up with any that do this at the start of a word. Does happen mid-word though.

**Comment by Jojo-Schmitz** [comment]
I blieve I found why the plugin doesn't work properly in Mu4, see https://github.com/dbolton/curlyQuotes/pull/2

**Comment by Thomas-Ganter** [comment]
> German keybords suffer from the same issue. So it is up to the software to auto-correct this. MS Office (Word, Outlook, etc.) does that, changing “bla bla” to „bla bla“ and ‘bla’ to ‚bla‘ There maybe colloquial German where those contractions are used, from the top of my head I can't come up with any that do this at the start of a word. Does happen mid- and end-word though (the latter e.g. for Genitive on words ending with an s or z in Nominativ already), but those don't pose a problem anway,
> 
> Note: the German close quotes are also different from the English ones, albeit rather subtle...
> 
> ![Image](https://github.com/user-attachments/assets/7af4f1be-62b6-4d8e-8f78-5d473f617d1f) 1st and 3rd pair English, 2nd and 4th pair German
> 
> See also https://en.wikipedia.org/wiki/Quotation_mark#Summary_table
> 
> See [dbolton/curlyQuotes#3](https://github.com/dbolton/curlyQuotes/pull/3)

@Jojo-Schmitz your Image is actually wrong.  
German double quotes are "ninety-nine, sixty-six", i.e. the closing german double quote is the same as the opening english double quote. See also the screenshot from your Wikipedia link 
<img width="674" height="68" alt="Image" src="https://github.com/user-attachments/assets/ce97b407-8e1e-417b-95b0-dc2b31de5167" />

Same for single-quotes.  

Is this really a Plugin Matter, as there are too many scenarios (like lyrics in multiple languages, or a different language than the locale of the UI, ...) or simply an input convenience question? 

Do not get my wrong: I am absolutely a warrior for correct typography. I even wrote a complaint to  _Der Spiegel_  about their incorrect use of quotation marks in their online articles. 

But I have also just learned to use the `option-^` and `shift-option-^` for german opening/closing quotes and `shift-option-^` and `shift-option-2` for english opening/closing quotes on my Mac.

**Comment by MRoth1910** [comment]
It's a lot of things.

The Curly Quotes plugin works now, but it's prone to breaking (there are some plugins that are rather useful and which MS 4 broke, and the maintainer stopped maintaining the plugin). It also only works on lyrics, but you have apostrophes and possibly quotation marks in all sorts of other kinds of text.

There's no reason which comes to mind for straight quotes in music publishing whatsoever. If there is, you should be able to enter that, e.g. via Uncode (entry of which is also broken; as far as I know, this bug which I reported a while ago hasn't been fixed, because it's still broken in 4.5.2), while leaving the smart/curly quotes as a default. The bottom line is that it's unprofessional to use the wrong punctuation mark, and if even MS Word can make this a default, other software programs should too.

So I would advocate for (2). I'm a bit late to the party, but since this isn't finished, I'd like to add my voice.

**Comment by Jojo-Schmitz** [comment]
Please reopen, the fix in #31460 is only half the story, works only for English style quotes, but not for German style quotes, as I mentioned in https://github.com/musescore/MuseScore/pull/31460#issuecomment-3664955083

**Comment by Jojo-Schmitz** [comment]
Actually #31525 seems to adress this, so the quotes will then follow the language setting of MuseScore.

**Comment by cbjeukendrup** [comment]
Yes, that's the idea. Eventually we might change it into a separate setting, and we're still hoping to someday decouple score language from UI language. But for now this is the easiest solution.

**Comment by Jojo-Schmitz** [comment]
Yes, score language is also my concern, a German text in a score on an English version of MuseScore should most probably show German quotes, and vice versa.

Then again there might even be a mishmash of German and English text in the same score (e.g. even lyrics in English, odd lyrics in German), so a score language wouldn't solve this. Only 'handmade' quotes would.
